scripts/ci/_all_libs.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# SCRIPTS_CI_DIR must be set
+
+if [[ "${SCRIPTS_CI_DIR}" == "" ]]; then
+        echo 2>&1
+        echo 2>&1 "Error - the script should have  SCRIPTS_CI_DIR variable set!"
+        echo 2>&1
+        exit 1
+fi
```

Comment:
FYI bash has something like this built in:

```suggestion
: "${SCRIPTS_CI_DIR:?this variable must be set before sourcing this file}"
```

Example usage:

```
❯ cat test.sh
set -euo pipefail
: "${SCRIPTS_CI_DIR:?this variable must be set before sourcing this file}"

❯ bash ./test.sh
./test.sh: line 2: SCRIPTS_CI_DIR: this variable must be set before sourcing this file
```

Reply:
That said, why don't we do it automatically?

```
 SCRIPTS_CI_DIR="$( dirname "${BASH_SOURCE[0]}" )"
```

Reply:
Nice. You learn every day :). 

I usally prefer to be a bit more explicit ifs rather than idiomatic bash constructs and add empty lines (they make error a bit more readable), but since this one provides also line number + script, it's much better. I will add it in couple more places :).

We need it because this will only works if we source that file from a bash script running in the "scripts/ci" directory. If we source it from ./breeze which is two levels up , the `$(dirname "${BASH_SOURCE[0]}"` returns the top level airflow sources directory.

Reply:
BASH_SOURCE[0] is the current file -- the equivalent of `__file__` in python:

```
airflow ❯ cat test.sh
. $( dirname "${BASH_SOURCE[0]}" )/tmp/test.sh

airflow ❯ cat tmp/test.sh
set -x
SCRIPTS_CI_DIR="$( dirname "${BASH_SOURCE[0]}" )"
echo BASH_SOURCE is "${BASH_SOURCE[@]}"


airflow ❯ bash -x test.sh
++ dirname ./test.sh
+ . ./tmp/test.sh
++ set -x
+++ dirname ./tmp/test.sh
++ SCRIPTS_CI_DIR=./tmp
++ echo BASH_SOURCE is ./tmp/test.sh ./test.sh
BASH_SOURCE is ./tmp/test.sh ./test.sh
```

And it works when run from somewhere else too:

```
airflow ❯ cd airflow && bash -x ../test.sh
++ dirname ../test.sh
+ . ../tmp/test.sh
++ set -x
+++ dirname ../tmp/test.sh
++ SCRIPTS_CI_DIR=../tmp
++ echo BASH_SOURCE is ../tmp/test.sh ../test.sh
BASH_SOURCE is ../tmp/test.sh ../test.sh

```

Reply:
(The `:` to do nothing, and the `${VAR:?Error message}` are both hard to understand until you know what they do, so I don't mind us not using those)

Reply:
> BASH_SOURCE[0] is the current file -- the equivalent of `__file__` in python:

Hmmm. Indeed. I was living under the impression than when you `source` the script, it is inlined in the original script. I had similar problems before but I think the reason was different - abusing _MY_DIR variable. I changed it. 

Thanks!. Again seems that with bash even after xxx years  you learn every day.

Reply:
And I am ok with :? - it gives line number which is really nice. And it's pretty obvious what it does if you start your message with "ERROR!".